server/src/main/java/org/opensearch/indices/replication/common/ReplicationCollection.java

```diff
@@ -310,7 +310,6 @@ public static class ReplicationRef<T extends ReplicationTarget> extends AutoClos
          */
         public ReplicationRef(T status) {
             super(status);
-            status.setLastAccessTime();
```

Comment:
In general I like this change. I would argue it's an anti-pattern to mutate the underlying object by simply constructing the wrapper. However, how can we have confidence that we're properly setting the last access time in all cases where necessary and not actually introducing a regression somewhere?

Reply:
Alright, adjusted the PR to handle timestamp updates centrally in createOrFinishListener.

Most recovery handlers go through createOrFinishListener, which now touches lastAccessTime only when handling a real request. The remaining explicit touches stay because those paths don’t use that helper:
- The initial doRecovery setup,
- The primary-context handoff (simple handoff + empty response)
- And the segment replication kickoff (local orchestration, not a transport handler)

 They’re still real work, so we touch them explicitly; ordinary polling via ReplicationCollection#get doesn’t bump the timestamp.